### PR TITLE
CHW/ANC SubTask - User can view a family profile

### LINKED ISSUE #284: CHW/ANC SubTask - User can view a family profile
- [x] Family profile details  - Members 
- [ ] Which family member has actual services due 
- [x] View Visit History  - Last Visit 2 or 3 dates (Encounters) - #290

image

**Acceptance criteria** 

- [x] User can be able to query for all members of a household from the Head of household's profile 
- [x] Use can be able to add a new member from the family profile 
- [x] User can be able to view the past 3 encounters with the ability to view more once clicking the see all button

### PR DESCRIPTION
**Pull request by syedowaisali90**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #284 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — COMMENTED** [review]
can we add tests for the new functionality here?

**Comment by syedowaisali90** [comment]
> can we add tests for the new functionality here?

Test cases added

File: android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  fun fetchEncounters(): LiveData<List<Encounter>> {
+    val data = MutableLiveData<List<Encounter>>()
+    CoroutineScope(dispatcherProvider.io()).launch {
+      val encounters =
+        fhirEngine.search<Encounter> {
+          filter(Encounter.SUBJECT) { value = "Patient/$familyId" }
+          from = 0
+          count = 3
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt:65]: what does this count mean?

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+import org.smartregister.fhircore.anc.R
+import org.smartregister.fhircore.anc.data.family.model.FamilyMemberItem
+import org.smartregister.fhircore.engine.ui.theme.AppTheme
+
+@Composable
+fun FamilyDetailScreen(dataProvider: FamilyDetailDataProvider) {
+
+  Surface(color = colorResource(id = R.color.white_smoke)) {
+    Column(
+      /*modifier = Modifier
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt:76]: should we remove this?

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+import org.hl7.fhir.r4.model.Encounter
+import org.hl7.fhir.r4.model.HumanName
+import org.hl7.fhir.r4.model.Patient
+import org.hl7.fhir.r4.model.Period
+import org.hl7.fhir.r4.model.StringType
+import org.smartregister.fhircore.anc.R
+import org.smartregister.fhircore.anc.data.family.model.FamilyMemberItem
+import org.smartregister.fhircore.engine.ui.theme.AppTheme
+
+@Composable
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt:71]: sweet is this using jetpack compose?

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt [hunk 4]
```diff
@@ -136,7 +136,7 @@ fun servicesCard(modifier: Modifier, text: String, color: Color) {
 @Composable
 @Preview(showBackground = true)
 fun FamilyRowPreview() {
-  val fmi = FamilyMemberItem("fm1", "21", "F", true)
+  val fmi = FamilyMemberItem("fmname", "fm1", "21", "F", true)
 
   val familyItem =
     FamilyItem("fid", "Name ", "M", "27", "Nairobi", true, listOf(fmi, fmi, fmi), 4, 5)
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt:142]: how are these used, is there a reason these need to be hard coded?

**Review by pld — COMMENTED** [review]
tests look great! a couple questions

File: android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt (hunk 1 above)
Comment (syedowaisali90) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt:65]: for now, we need to show only 3 encounters and we have a `See All` button for the encounter list

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt (hunk 2 above)
Comment (syedowaisali90) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt:76]: done.

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt (hunk 3 above)
Comment (syedowaisali90) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreen.kt:71]: yes

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt (hunk 4 above)
Comment (syedowaisali90) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt:142]: it's just being used for preview in the editor

Comment (pld) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt:142]: k cool, sorry I'm not that familiar with this, what does that mean?

Comment (syedowaisali90) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt:142]: on designing a UI in we need to see how the UI will look on an actual device/emulator so, for that purpose Android Studio (IDE) provides a layout preview window. here is an example https://developer.android.com/jetpack/compose/tooling

Comment (pld) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/family/register/components/FamilyListItem.kt:142]: got ok sounds good, thank you for explaining

**Review by pld — DISMISSED** [review]

File: android/anc/src/androidTest/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreenTest.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.smartregister.fhircore.engine.ui.theme.AppTheme
+
+class FamilyDetailScreenTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Ignore("will be worked later")
```
Comment (ndegwamartin) [inline android/anc/src/androidTest/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreenTest.kt:29]: I think we can remove this test class, integration testing is not working yet

File: android/anc/build.gradle [hunk 6]
```diff
@@ -151,4 +153,7 @@ dependencies {
     androidTestImplementation deps.atsl.ext_junit
     androidTestImplementation deps.atsl.espresso
     implementation deps.work.runtime
+
+    androidTestImplementation("androidx.compose.ui:ui-test-junit4:1.0.1")
```
Comment (ndegwamartin) [inline android/anc/build.gradle:157]: Currently on Unit tests supported on CI. BDD Gherkin tests are being implemented separately for ui/integration

File: android/anc/src/androidTest/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreenTest.kt (hunk 5 above)
Comment (syedowaisali90) [reply android/anc/src/androidTest/java/org/smartregister/fhircore/anc/ui/family/details/FamilyDetailScreenTest.kt:29]: done.

File: android/anc/build.gradle (hunk 6 above)
Comment (syedowaisali90) [reply android/anc/build.gradle:157]: done

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**